pkg/apis/allocation/v1/gameserverallocation.go

```diff
@@ -279,6 +279,7 @@ type GameServerAllocationStatus struct {
 	Ports          []agonesv1.GameServerStatusPort `json:"ports,omitempty"`
 	Address        string                          `json:"address,omitempty"`
 	NodeName       string                          `json:"nodeName,omitempty"`
+	Remote         bool                            `json:"Remote"`
```

Comment:
I had a thought on this today. Should this be:

```suggestion
	Source         string                            `json:"source"`
```

?

Where the source is populated with the url details the remote call came from? (maybe defaults to "local").

This would help solve some of the problems faced by #2396, and also provide more insight into multi cluster allocation when debugging. WDYT?

(Also, json field are `lowerCase`)

Reply:
Reading through the issue, I think as a start, it makes sense the expose the remote allocation endpoint in the `GameServerAllocationStatus` and `AllocationResponse` to indicate the source of the allocation. Made the change accordingly. We can iterate more on whether to expose label/annotation, and whether they should be exposed behind feature flag.